io.openems.edge.controller.ess.timeofusetariff/src/io/openems/edge/controller/ess/timeofusetariff/TimeOfUseTariffController.java

```diff
@@ -9,58 +9,63 @@
 import io.openems.edge.common.component.OpenemsComponent;
 import io.openems.edge.controller.api.Controller;
 
-public interface ControllerEssTimeOfUseTariffDischarge extends Controller, OpenemsComponent {
+public interface TimeOfUseTariffController extends Controller, OpenemsComponent {
 
 	public enum ChannelId implements io.openems.edge.common.channel.ChannelId {
-
 		/**
-		 * Current state of the Time of use tariff discharge controller.
+		 * Current state of the Time of use tariff controller.
 		 */
 		STATE_MACHINE(Doc.of(StateMachine.values()) //
+				.persistencePriority(PersistencePriority.VERY_HIGH) //
 				.text("Current state of the Controller")),
 
+		/**
+		 * Aggregated seconds when storage is being force charged.
+		 */
+		CHARGED_TIME(Doc.of(OpenemsType.LONG) //
+				.unit(Unit.CUMULATED_SECONDS) //
+				.persistencePriority(PersistencePriority.HIGH)), //
+		CHARGED(Doc.of(OpenemsType.BOOLEAN)//
+				.text("The controller currently recharge the battery")),
+		CHARGE_VALUE(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT)//
+				.text("The amount of energy set to charge from grid in a period")), //
+
 		/**
 		 * Aggregated seconds when storage is blocked for discharge.
 		 */
 		DELAYED_TIME(Doc.of(OpenemsType.LONG) //
 				.unit(Unit.CUMULATED_SECONDS) //
 				.persistencePriority(PersistencePriority.HIGH)), //
-
 		DELAYED(Doc.of(OpenemsType.BOOLEAN)//
 				.text("The controller currently blocks discharge")),
-		TARGET_HOURS_IS_EMPTY(Doc.of(OpenemsType.BOOLEAN)//
-				.text("The list of target hours is empty")),
-		QUATERLY_PRICES_TAKEN(Doc.of(OpenemsType.BOOLEAN)//
-				.text("The controller retrieves hourly Prices from API successfully")),
-		TARGET_HOURS_CALCULATED(Doc.of(OpenemsType.BOOLEAN)//
-				.text("The controller calculates target time to buy from grid successfully")),
-		TOTAL_CONSUMPTION(Doc.of(OpenemsType.INTEGER) //
-				.text("Total consmption for the night")),
-		QUARTERLY_PRICES(Doc.of(OpenemsType.FLOAT) //
-				.unit(Unit.EUROS_PER_MEGAWATT_HOUR) //
-				.text("Price of the electricity for the current Hour")),
-		REMAINING_CONSUMPTION(Doc.of(OpenemsType.DOUBLE) //
-				.text("remaining consmption to charge from grid")),
-		TARGET_HOURS(Doc.of(OpenemsType.INTEGER) //
-				.text("Number of Target Hours")),
+		DISCHARGE_VALUE(Doc.of(OpenemsType.INTEGER) //
+				.unit(Unit.WATT)//
+				.text("The amount of energy set to discharge from the battery in a period")), //
+
+		CHARGE_DISCHARGE_ENERGY(Doc.of(OpenemsType.INTEGER) //
+				.text("Charge/Discharge energy calculated for the period.")), //
```

Comment:
So this is the same as CHARGE_VALUE and DISCHARGE_VALUE?

Please remove unnecessary channels. 

Be aware that any Channel that is required for UI has to have a persistencePriority of HIGH.